core/tests/protractor_desktop/coreEditorAndPlayerFeatures.js

```diff
@@ -110,6 +110,7 @@ describe('Enable correctness feedback and set correctness', function() {
     // Go back to mark the solution as correct.
     explorationEditorPage.navigateToMainTab();
     explorationEditorMainTab.moveToState('First');
+    waitFor.visibilityOf(element(by.css('.protractor-test-response-tab')));
```

Comment:
I would suggest moving this to the protractor_utils file of `explorationEditorMainTab` in the `getResponseEditor` method. Utils files have code to catch and test for something to happen. These test codes are more readable and expect util services to do all the wait and catch stuff. (+ this will make `getResponseEditor` more robust and fixing flakiness in all the places which uses this method.)